markdown/3.x/en/ff-communication.md

```diff
@@ -27,4 +27,17 @@ correctly.
 
 **NOTE**: Although `version` defaults to `7.2` when not set, it is recommended to set it as default value may change with feature releases.
 
-For more information, see the [API reference](/api/3.x/ff-communication#tab=api).
\ No newline at end of file
+For more information, see the [API reference](/api/3.x/ff-communication#tab=api).
+
+### Single hit redirect deprecation
+Due to limitations this functionality brings, it will be removed in future releases.
+In order to prevent infinite redirection after getting back from product page to search results page, an internal `ff-no-redirect` flag is toggled after the first redirection
+takes place. With this flag toggled Web Components will not redirect next search, even if there is only one record returned back from the FACT-Finder. However, the next search 
+will not necessarily be the one, issued after getting back to the search results page. If user provides `ff-searchbox` component on the product page, the next search could
+also be issued from there. Single hit redirect in this case will not work, because of `ff-no-redirect` flag is toggled on and in result user will land on the search result page.
```

Comment:
```suggestion
In order to prevent infinite redirection after getting back from a product page to the search results page, an internal `ff-no-redirect` flag is toggled after the first redirection takes place.
With this flag toggled Web Components will not redirect the next search, even if there is only one record returned back from FACT-Finder.
However, the next search will not necessarily be the one that was issued after returning to the search results page.
If the user provides an `ff-searchbox` component on the product page, the next search could also be issued from there.
Single hit redirect in this case will not work because of the `ff-no-redirect` flag is toggled on and as a result the user will land on the search result page.
```

Reply:
changed